packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -170,21 +131,16 @@ export class DebugSession implements Disposable {
       .then(() => disposeAll(this.disposables));
   }
 
-  public async startJSDebugSession(metro: Metro, extraConfiguration?: DebugExtraConfiguration) {
+  public async startJSDebugSession(configuration: JSDebugConfiguration) {
     if (this.jsDebugSession) {
       await this.restart();
     }
 
-    const websocketAddress = extraConfiguration?.websocketAddress || (await metro.getDebuggerURL());
-    if (!websocketAddress) {
-      return false;
-    }
-
-    const isUsingNewDebugger = metro.isUsingNewDebugger;
+    const isUsingNewDebugger = configuration.isUsingNewDebugger;
     const debuggerType = isUsingNewDebugger ? PROXY_JS_DEBUGGER_TYPE : OLD_JS_DEBUGGER_TYPE;
 
     const sourceMapPathOverrides: Record<string, string> = {};
-    const metroWatchFolders = metro.watchFolders;
+    const metroWatchFolders = configuration.watchFolders;
```

Comment:
NIT: _In my opinion_, we should pass `sourceMapPathOverrides` in the `configuration` here, and calculate them from `watchFolders` either in `Metro` or wherever we call `startJSDebugSession` (i.e. `deviceSession`), but that's not very important.